File: app/src/main/java/pusios/com/soundfy/db/DbBuilder.java
package pusios.com.soundfy.db;

import com.google.gson.Gson;

import java.util.ArrayList;
import java.util.List;

import pusios.com.soundfy.model.Author;
import pusios.com.soundfy.model.Catalog;
import pusios.com.soundfy.model.Clip;
import pusios.com.soundfy.model.Party;

public class DbBuilder {

    private Author author;
    private Party party;
    private int catalogIdCounter;

    public String buildDb(final Gson gson) {
        return gson.toJson(createCatalog());
    }

    private Catalog createCatalog() {
        final List<Catalog> catalogs = new ArrayList<>();

        author = createAuthor("rajoy");
        author.getClips().add(new Clip("raw.rajoy.hola", "hola"));
        author.getClips().add(new Clip("raw.rajoy.adios", "adios"));

        party = createParty("pp", "blue");
        party.getSubCatalogs().add(author);
        catalogs.add(party);

        author = createAuthor("iglesias");
        author.getClips().add(new Clip("raw.iglesias.si", "si"));
        author.getClips().add(new Clip("raw.iglesias.no", "no"));

        party = createParty("podemos", "morado");
        party.getSubCatalogs().add(author);
        catalogs.add(party);

        return new Catalog(0, catalogs, new ArrayList<Clip>());
    }

    private Party createParty(final String name, final String color) {
        return new Party(++catalogIdCounter,
                         name,
                         color,
                         new ArrayList<Catalog>(),
                         new ArrayList<Clip>());
    }

    private Author createAuthor(final String name) {
        return new Author(++catalogIdCounter,
                          name,
                          new ArrayList<Catalog>(),
                          new ArrayList<Clip>());
    }
}

File: app/src/main/java/pusios/com/soundfy/db/DbBuilder.kt
package pusios.com.soundfy.db

import com.google.gson.Gson

import java.util.ArrayList

import pusios.com.soundfy.model.Author
import pusios.com.soundfy.model.Catalog
import pusios.com.soundfy.model.Clip
import pusios.com.soundfy.model.Party

class DbBuilder {

    private lateinit var author: Author
    private lateinit var party: Party
    private var catalogIdCounter: Int = 0

    fun buildDb(gson: Gson): String {
        return gson.toJson(createCatalog())
    }

    private fun createCatalog(): Catalog {

        val clips = mutableListOf<Clip>()
        clips.add(Clip("bodyfat", "body fat"))
        clips.add(Clip("hasta", "hasta la vista"))
        clips.add(Clip("uglymf", "ugly mf"))
        clips.add(Clip("wakeupdrunk", "wake up"))
        clips.add(Clip("wtfgoinon", "wtf"))

        /*val catalogs = ArrayList<Catalog>()

        author = createAuthor("rajoy")
        author.clips.add(Clip("raw.rajoy.hola", "hola"))
        author.clips.add(Clip("raw.rajoy.adios", "adios"))

        party = createParty("pp", "blue")
        party.subCatalogs.add(author)
        catalogs.add(party)

        author = createAuthor("iglesias")
        author.clips.add(Clip("raw.iglesias.si", "si"))
        author.clips.add(Clip("raw.iglesias.no", "no"))

        party = createParty("podemos", "morado")
        party.subCatalogs.add(author)
        catalogs.add(party)

        return Catalog(0, catalogs, ArrayList<Clip>())*/


        return Catalog(0, mutableListOf(), clips)
    }

    private fun createParty(name: String, color: String): Party {
        return Party(++catalogIdCounter,
                name,
                color,
                ArrayList<Catalog>(),
                ArrayList<Clip>())
    }

    private fun createAuthor(name: String): Author {
        return Author(++catalogIdCounter,
                name,
                ArrayList<Catalog>(),
                ArrayList<Clip>())
    }
}
